packages/altair-app/src/app/modules/altair/utils/index.ts

```diff
@@ -75,8 +75,33 @@ export const openFile = async (opts: FileDialogOptions = {}) => {
   } catch (err) {
     debug.log('There was an issue while opening the file: ', err);
   }
+};
+
+export const getFileContent = async (file: File) => {
+  return new Promise<string>((resolve, reject) => {
+    const fileReader = new FileReader();
+    fileReader.onload = function (e: any) {
+      const contents: string = e.target.result;
+
+      // Resolve file content
+      resolve(contents);
+    };
+    fileReader.readAsText(file);
+  })
 }
 
+export const openFiles = async (opts: FileDialogOptions = {}) => {
+  try {
+    const files = await fileDialog({ ...opts, multiple: true });
+
+    return Promise.all(Array.from(Array(files.length)).map((_, i) => getFileContent(files[i])));
```

Comment:
Bumping this comment in case you missed it

> Really? And you tried with [...files]? I know that FileList is merely an array-like object but I've used this pattern a few times now and I'm not sure why it wouldn't work for you. Also just tested it quickly in [jsfiddle](https://jsfiddle.net/imolorhe/t237ushc/) and seems to work. I think it makes for cleaner and simpler code here, which is why I'm asking for that instead